src/main/java/net/minecraftforge/items/IItemHandler.java

```diff
@@ -0,0 +1,49 @@
+package net.minecraftforge.items;
+
+import net.minecraft.item.ItemStack;
+
+public interface IItemHandler
+{
+    /**
+     * Returns the number of slots available
+     *
+     * @return The number of slots available
+     **/
+    int getSlots();
+
+    /**
+     * Returns the ItemStack in a given slot.
+     * The resultant value's stacksize may be greater than the stacks max size.
+     * <p/>
+     * IMPORTANT: This ItemStack MUST NOT be modified. This method is not for
+     * altering an inventories contents. Any implementers who are able to detect
+     * modification through this method should throw an exception.
+     * <p/>
+     * SERIOUSLY: DO NOT MODIFY
+     *
+     * @param slot Slot to query
+     * @return ItemStack in given slot. May be null.
+     **/
+    ItemStack getStackInSlot(int slot);
```

Comment:
Neglected to consider NBT in that relation, though this could be faked using some interfaces. In which we have getter methods for NBT data but not setter methods.